pkg/apis/build/v1alpha2/build.go

```diff
@@ -137,8 +137,7 @@ func (b *Build) PodName() string {
 }
 
 func (b *Build) MetadataReady(pod *corev1.Pod) bool {
-	return !b.Status.GetCondition(corev1alpha1.ConditionSucceeded).IsTrue() &&
-		pod.Status.Phase == "Succeeded"
+	return pod.Status.Phase == "Succeeded" || (pod.Status.Phase == "Failed" && pod.Status.Reason == "DeadlineExceeded")
```

Comment:
If you're interested in using constants for phase values, they're defined on k8s.io/api/core/v1: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L2393-L2414